python/tvm/relay/frontend/onnx.py

```diff
@@ -4008,6 +4008,22 @@ def _impl_v1(cls, inputs, attr, params):
         for var in else_free_vars:
             graph_scope._nodes.update({var.name_hint: var})
 
+        # Sometimes pytorch to onnx will insert silly if statements that produce dynamic ranks.
+        # Often these dont contribute anything. If we see a dynamic rank output, try to unify
+        # them so we can continue without breaking.
+        then_shape = infer_shape(then_expr)
```

Comment:
Very sus, but I guess this is the best we can do in this situation. The only reason I will allow this is it's in converting `If` which is inherently going to be a bit more sus due to relay limitations.

Reply:
@ehsanmok noted that he's working on exposing Optional types in relay that may allow a more elegant solution once it lands.